codenames-domain/src/main/java/com/codenames/repository/UserRepository.java

```diff
@@ -0,0 +1,11 @@
+package com.codenames.repository;
+
+import com.codenames.entity.UserEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+
+public interface UserRepository extends JpaRepository<UserEntity, Long> {
+
+    <R> Optional<R> findByLogin(Class<R> type, String nickname);
```

Comment:
Why do you need it to be generic? Also use `<T>` for type

Reply:
So I redid it. I called it a generic because I meant that it was a type of return. And the generic one is just for convenience so that you can take any type of user data